src/Microsoft.Win32.Registry/src/Microsoft/Win32/Registry.cs

```diff
@@ -77,41 +76,39 @@ private static RegistryKey GetBaseKeyFromKeyName(string keyName, out string subK
                 throw new ArgumentNullException("keyName");
             }
 
-            string basekeyName;
-            int i = keyName.IndexOf('\\');
-            if (i != -1)
+            RegistryKey basekey = null;
+            const StringComparison comparisonType = StringComparison.OrdinalIgnoreCase;
+
+            if (keyName.StartsWith("HKEY_CURRENT_USER", comparisonType))
```

Comment:
This doesn't look like an equivalent transformation.  If I pass in a keyName of "HKEY_CURRENT_USER_STEPHEN", previously this would throw an exception, but now it's going to match HKEY_CURRENT_USER at the start, set baseKey to Registry.CurrentUser, and set subKeyName to null.